<launch>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <!-- <rosparam file="$(find flowerbot_control)/config/flowerbot_control.yaml" command="load"/> -->

    <!-- <node name="generate_configuration" pkg="flowerbot_control" type="generate_configuration" respawn="false" output="screen" ns="/flowerbot" args="$(find robot_base)/robot_config/4wd4ws_config.yaml $(find flowerbot_control)/config/flowerbot_control.yaml"/> -->


    <!-- load the controllers -->
    <!-- <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/flowerbot" args="wheel0_steering_joint_control joint_state_controller"/>   -->
    <!-- Load the URDF into the ROS Parameter Server -->
    <arg name="urdf_file" default="$(find xacro)/xacro --inorder '$(find flowerbot_description)/urdf/flowerbot.urdf.xacro' configuration_file:=$(find robot_base)/robot_config/2wd2ws_config.yaml" />
    <param name="robot_description" command="$(arg urdf_file)" />

    <node name="robot_drive" pkg="robot_drive" type="robot_drive" output="screen">
        <param name="wheel_driver" value="kangaroo_driver/KangarooDriver"/>
        <param name="robot_controller" value="robot_base/BasicRobotController"/>
        <param name="robot_config_file" value="$(find robot_base)/robot_config/2wd2ws_config.yaml"/>
        <param name="odom_publish_rate" value="25"/>
        <param name="simulation" value="false"/>
        <rosparam command="load" file="$(find kangaroo_driver)/config/kangaroo_driver.yaml"/>
        <!-- <rosparam command="load" file="$(find robot_base)/config/kangaroo_driver.yaml"/> -->
    </node>
    <!-- convert joint states to TF transforms for rviz, etc -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <remap from="/joint_states" to="/robot_drive/joint_states" />
    </node>
    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="dev" value="/dev/control_joystick" />
        <param name="deadzone" value="0.05" />
        <param name="autorepeat_rate" value="20" />
    </node>

    <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
        <param name="enable_button" value="4" />
		<param name="enable_turbo_button" value="5" />
        <param name="axis_linear" value="4" />
        <param name="scale_linear" value="0.3" />
		<param name="scale_linear_turbo" value="1.5" />
        <param name="axis_angular" value="3" />
        <param name="scale_angular" value="0.5" />
    </node>
</launch>
